services/user.py

```diff
@@ -0,0 +1,36 @@
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import User
+
+def create_user(username: str, password: str, email: str = None, first_name: str = None, last_name: str = None) -> User:
+    user = get_user_model().objects.create_user(username=username, password=password)
+
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+
+    user.set_password(password)
+    user.save()
+
+    return user
+
+def get_user(user_id: int) -> User:
+    return get_user_model().objects.get(pk=user_id)
+
+def update_user(user_id: int, username: str = None, password: str = None, email: str = None, first_name: str = None, last_name: str = None) -> None:
+    user = get_user(user_id)
+
+    if username:
+        user.username = username
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+    if password:
+        user.set_password(password)
+
+    user.save()

```

Comment:
This function updates the user's attributes and then saves the user. This can lead to the user being saved even if no attributes were changed. It would be more efficient to check if any attributes were changed before calling `save()`. You could do this by adding a boolean flag that is set to true if any attribute is changed, and then only calling `save()` if the flag is true.